blockchain/encoder_initializer.go

```diff
@@ -0,0 +1,34 @@
+package blockchain
+
+import (
+	"reflect"
+	"sync"
+
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/encoder"
+)
+
+var once sync.Once
+
+func registerCoreTypesToEncoder() {
+	registerAndPanicOnError := func(t reflect.Type) {
+		err := encoder.RegisterType(t)
+		if err != nil {
+			panic(err)
+		}
+	}
```

Comment:
I think we can move the function body to line 31 and remove the function call.

Would it make more sense to have the registration happen on a call to `pebble.New`? I think that would allow us to remove `sync.Once` magic and keep database-specific encodings separate from the blockchain. I'm not sure if that's the right way to go and would appreciate your take.

Reply:
We need to register some of our types because it is a limitation of our encoder library. DB doesnt really care about the data it is storing. For DB they are all byte arrays. 

It made more sense to me to be in blockchain since it is the one that actually uses the `encoder` package. For example; if someone uses our blockchain struct only as a library in their application and implement our `DB` interface themselves. They would still expect to be able to encode/decode these types fine.

`sync.Once` is only there because I wanted to allow multiple `blockchain` structs to be instantiated, without additional calls to `RegisterType`. `RegisterType` is stateful, so it is important that we call it once per type.

Reply:
That makes sense. Can we inline `registerAndPanicOnError`?